cmd/storage-rest-client.go

```diff
@@ -398,19 +405,47 @@ func (client *storageRESTClient) RenameData(ctx context.Context, srcVolume, srcP
 	return err
 }
 
-func (client *storageRESTClient) ReadVersion(ctx context.Context, volume, path, versionID string) (fi FileInfo, err error) {
+// where we keep old *Readers
+var readMsgpReaderPool = sync.Pool{New: func() interface{} { return &msgp.Reader{} }}
```

Comment:
```suggestion
var readMsgpReaderPool = sync.Pool{New: func() interface{} { return msgp.NewReaderSize(8<<10) }}
```

Reply:
Yeah well I avoided this to do it very similar to what msgp/ library is doing this. https://github.com/tinylib/msgp/blob/master/msgp/read.go#L13